packages/permissions/src/PermissionsAPI.ts

```diff
@@ -0,0 +1,362 @@
+ * @link [Browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API#browser_compatibility)
+ *
+ * @class PermissionsAPI
+ */
+export class PermissionsAPI {
+  /**
+   * A reference to the Permissions API provided by the browser.
+   * @private
+   * @static
+   * @type {Permissions}
+   */
+  static #permissions: Permissions = navigator.permissions
+
+  /**
+   * An object to store the current state of permissions.
+   * @private
+   * @static
+   * @type {Object<string, PermissionResponse>}
+   */
+  static #permissionStates: {
+    [key: string]: PermissionResponse
+  } = {}
+
+  /**
+   * An object to manage event listeners for permission changes.
+   * @private
+   * @static
+   * @type {EventsLogs}
+   */
+  static #events: EventsLogs = {}
+
+  /**
+   * Retrieves a handler for synchronous permission requests.
+   * @private
+   * @static
+   * @param {PermissionOption} permissionOption - The permission option to request.
+   * @param {PermissionHandlerOption} [handlerOption] - Optional handler options for granted/denied/error callbacks.
+   * @returns {IPermissionHandler} - A permission handler for synchronous requests.
+   */
+  static #getHandler(
+    permissionOption: PermissionOption,
+    handlerOption?: PermissionHandlerOption
+  ): IPermissionHandler {
+    const _handler: IPermissionHandler = function (): void {
+      const _events = PermissionsAPI.#events[_handler.eventId]
+
+      PermissionsAPI.getPermission(permissionOption)
+        .then((_permission) => {
+          if (_events.onPermissionChange) {
+            _permission.status.onchange = function () {
+              _events.onPermissionChange({
+                name: this.name,
+                status: this,
+                state: this.state
+              })
+            }
+          }
+
+          if (_permission.state === 'denied') {
+            if (_events.onPermissionDenied)
+              _events.onPermissionDenied(
+                _permission as PermissionResponse<'denied'>
+              )
+            if (handlerOption?.denied)
+              handlerOption.denied(_permission as PermissionResponse<'denied'>)
+          } else {
+            if (_events.onPermissionGranted)
+              _events.onPermissionGranted(
+                _permission as PermissionResponse<'granted'>
+              )
+            if (handlerOption?.granted)
+              handlerOption.granted(
+                _permission as PermissionResponse<'granted'>
+              )
+          }
+        })
+        .catch((error) => {
+          if (_events.onPermissionError)
+            _events.onPermissionError(error as PermissionResponse<'error'>)
+          if (handlerOption?.error)
+            handlerOption.error(error as PermissionResponse<'error'>)
+        })
+    }
+    _handler.eventId = crypto.randomUUID()
+    PermissionsAPI.#events[_handler.eventId] = {}
+
+    return _handler
+  }
```

Comment:
The function getHandler does not handle the case where navigator.permissions is null, which would cause runtime errors.

```diff
-    const _handler: IPermissionHandler = function (): void {
+    if (!PermissionsAPI.#permissions) {
+      throw new Error('Permissions API is not supported in this environment.');
+    }
+
+    const _handler: IPermissionHandler = function (): void {
```